redpanda/resources/cluster/resource_cluster.go

```diff
@@ -168,30 +166,26 @@ func (c *Cluster) Create(ctx context.Context, req resource.CreateRequest, resp *
 		resp.Diagnostics.AddError("unable to parse CreateCluster request", err.Error())
 		return
 	}
-	op, err := c.CluClient.CreateCluster(ctx, &controlplanev1beta1.CreateClusterRequest{
-		Cluster: clusterReq,
-	})
+	clResp, err := c.CpCl.Cluster.CreateCluster(ctx, &controlplanev1beta2.CreateClusterRequest{Cluster: clusterReq})
 	if err != nil {
 		resp.Diagnostics.AddError("failed to create cluster", err.Error())
 		return
 	}
-	if _, err := utils.GetClusterUntilRunningState(ctx, 0, 50, clusterReq.Name, c.CluClient); err != nil {
+	if _, err := utils.GetClusterUntilRunningState(ctx, 0, 80, clusterReq.Name, c.CpCl); err != nil {
 		resp.Diagnostics.AddError("failed at getting ready state while creating cluster", err.Error())
 		return
 	}
-
-	var metadata controlplanev1beta1.CreateClusterMetadata
+	op := clResp.Operation
+	var metadata controlplanev1beta2.CreateClusterMetadata
 	if err := op.Metadata.UnmarshalTo(&metadata); err != nil {
 		resp.Diagnostics.AddError("failed to unmarshal cluster metadata", err.Error())
 		return
 	}
-	if err := utils.AreWeDoneYet(ctx, op, 60*time.Minute, time.Minute, c.OpsClient); err != nil {
+	if err := utils.AreWeDoneYet(ctx, op, 60*time.Minute, time.Minute, c.CpCl.Operation); err != nil {
```

Comment:
Since we are passing a context, I was sort of expecting the deadline to be set there. So, it cascades releasing up resources when the operation is cancelled.